backend/de.metas.contracts/src/main/java/de/metas/contracts/impl/FlatrateBL.java

```diff
@@ -1446,9 +1451,10 @@ private I_C_Flatrate_Transition getTransitionForTerm(@NonNull final I_C_Flatrate
 	private void updateEndDate(final I_C_Flatrate_Transition transition, final I_C_Flatrate_Term term)
 	{
 		final Timestamp endDate = computeEndDate(transition, term);
-		term.setEndDate(endDate);
+		term.setEndDate(endDate == null ? term.getEndDate() : endDate);
```

Comment:
IMHO this logic shall go into computeEndDate, so the returned endDate shall be already OK. wdyt?